rfcs/0018-glides-trainsheet-events.md

```diff
@@ -82,6 +84,7 @@ Information about how the event was created by Glides. All important outputs fro
 
 Fields:
 - `author` ([GlidesUser](#GlidesUser), optional): The logged-in user whose action triggered the event.
+- `inputTimestamp` (RFC3999 timestamp, optional): The timestamp that the user entered the data into Glides, as reported by their device. This is different from the `time` field in the event, which is the server-reported time that the Glides server received the input. They could differ if sending the input was delayed do to network problems, or if the device's clock is wrong. Usually, you will want the event time instead, because Glides applies changes in the order they were received.
```

Comment:
I think this is backwards. `time` in the event should be when the event happened, and you can add an additional `receivedTime` metadata field for when Glides learned about it.

Reply:
There are a couple main reasons to use the received time as the primary event time:
- It's generated by the client device, and so is in some sense less trustworthy. What if a client's clock is set wrong? We don't do any validation, and we can't, because we can't tell the difference between an incorrect time and a correct time that was delayed by network issues. This isn't so much a specific concern as a general practice of not depending on non-validated client data.
- Glides applies updates in the order they're received. If the received timestamp is used as the event time, then applying events in order by their timestamp will give you the same state as in Glides. If we use the client timestamp, then if an event is delayed so it's received after another later input, a client who checks the event timestamp could apply them in the wrong order and end up with a different result than Glides. This could be mitigated by documenting that the client-provided event timestamp is unreliable, but that's easier to document if it's included with other input metadata here.
- The CloudEvents [docs](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#time) seem to say the time field's consistency is more important than it being original. It'll be much easier to stay consistent with server-generated timestamps.

Reply:
Okay, sounds good to me. This is good context that should go into the RFC somewhere.

Reply:
Done https://github.com/mbta/technology-docs/pull/18/commits/12ba9115deac3dcc5e9eac484d9c30ab6048b394